src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/samples/RESTSampleSequenceFilesController.java

```diff
@@ -568,7 +575,14 @@ public ModelMap addNewSequenceFilePairToSample(@PathVariable Long sampleId,
 			modelMap.addAttribute(RESTGenericController.RESOURCE_NAME, sequencingObject);
 		} catch (IllegalArgumentException e) {
 			logger.debug("Error 400 - Bad Request: " + e.getMessage());
-			response.setStatus(HttpStatus.BAD_REQUEST.value());
+			throw e;
+		} finally {
+			// clean up the temporary files.
+			Files.deleteIfExists(target1);
+			Files.deleteIfExists(temp1);
+			Files.deleteIfExists(target2);
+			Files.deleteIfExists(temp2);
+			logger.trace("Deleted temp files");
```

Comment:
Can you move this logging statement above the `Files.deleteIfExists`?  Better to log before you do something so you know what's happening if something errors out.

Reply:
Changed. Also added a log statement to the single end file deletion too.